File: backend/auth-service/src/main/java/com/farm2home/auth/service/AuthService.java
package com.farm2home.auth.service;

import com.farm2home.auth.domain.entity.User;
import com.farm2home.auth.dto.request.*;
import com.farm2home.auth.dto.response.AuthResponse;
import com.farm2home.auth.dto.response.GoogleAuthResponse;
import com.farm2home.auth.dto.response.OtpVerifyResponse;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    void sendOtp(OtpRequest request);

    /** @return registrationRequired=false with auth=null for REGISTRATION/FORGOT_PASSWORD
     *  (unchanged behavior); for LOGIN, auth populated (existing account, signed in) or
     *  registrationRequired=true (no account for this mobile yet - continue registration). */
    OtpVerifyResponse verifyOtp(VerifyOtpRequest request);

    /** Validates the Google credential server-side and either signs the caller into an existing/
     *  just-linked account, or reports registrationRequired=true so the frontend can continue into
     *  the existing registration wizard - see AUTH_SOCIAL_OTP_PROGRESS.md's account-linking policy. */
    GoogleAuthResponse googleAuth(GoogleAuthRequest request);

    AuthResponse refreshToken(RefreshTokenRequest request);

    void logout(String accessToken);

    /** Maps the already-authenticated principal (resolved by JwtAuthenticationFilter from the
     *  bearer token) to the same UserInfo shape login/register return - lets the frontend
     *  repopulate its in-memory user object (username, roles, etc.) on app boot/page refresh,
     *  since only the token itself, not that object, survives a reload. */
    AuthResponse.UserInfo getCurrentUserInfo(User user);
}

File: frontend/src/services/authService.ts
import axiosClient from './authAxiosClient'
import type {
  LoginRequest, RegisterRequest, OtpRequest, OtpType, AuthResponse, UserInfo,
  VerifyOtpResponse, GoogleAuthRequest, GoogleAuthResponse,
} from '../types/auth.types'
import type { ApiResponse } from '../types/common.types'

const AUTH_BASE = '/auth'

export const authService = {
  login: (data: LoginRequest) =>
    axiosClient.post<ApiResponse<AuthResponse>>(`${AUTH_BASE}/login`, data),

  register: (data: RegisterRequest) =>
    axiosClient.post<ApiResponse<AuthResponse>>(`${AUTH_BASE}/register`, data),

  sendOtp: (data: OtpRequest) =>
    axiosClient.post(`${AUTH_BASE}/send-otp`, data),

  // For otpType='LOGIN' this IS Mobile OTP Login: response.data.auth is populated (existing
  // account, signed in) or response.data.registrationRequired is true (no account yet - continue
  // registration, see MobileOtpDialog). For REGISTRATION/FORGOT_PASSWORD, response.data.auth is
  // always undefined - identical observable behavior to before this endpoint's response gained a
  // body.
  verifyOtp: (mobile: string, otp: string, otpType: OtpType) =>
    axiosClient.post<ApiResponse<VerifyOtpResponse>>(`${AUTH_BASE}/verify-otp`, { mobile, otp, otpType }),

  // Validates the Google credential server-side. response.data.auth populated → signed in;
  // response.data.registrationRequired → no Farm2Home account yet, continue into RegisterPage
  // (see LoginPage.tsx's handleGoogleCredentialResponse).
  googleAuth: (data: GoogleAuthRequest) =>
    axiosClient.post<ApiResponse<GoogleAuthResponse>>(`${AUTH_BASE}/google`, data),

  refreshToken: (refreshToken: string) =>
    axiosClient.post<ApiResponse<AuthResponse>>(`${AUTH_BASE}/refresh-token`, { refreshToken }),

  logout: () =>
    axiosClient.post(`${AUTH_BASE}/logout`),

  // Only the token survives a page reload (see authSlice's initialState) - this repopulates the
  // in-memory user object (username, roles, etc.) from it on app boot.
  getMe: () =>
    axiosClient.get<ApiResponse<UserInfo>>(`${AUTH_BASE}/me`),
}
